clustered/server/src/main/java/org/ehcache/clustered/server/offheap/OffHeapChainStorageEngine.java

```diff
@@ -285,24 +285,26 @@ public boolean removeHeader(Chain header) {
       Iterator<Element> iterator = header.iterator();
       do {
         if (!compare(iterator.next(), suffixHead)) {
-          return false;
+          return true;
         }
         prefixTail = suffixHead;
         suffixHead = storage.readLong(suffixHead + ELEMENT_HEADER_NEXT_OFFSET);
       } while (iterator.hasNext());
 
       if (suffixHead == chain) {
         //whole chain removed
-        evict();
+        int slot = owner.getSlotForHashAndEncoding(readKeyHash(chain), chain, ~0);
+        if (!owner.evict(slot, true)) {
+          throw new AssertionError();
```

Comment:
I presume this is a "should not happen" case but is there something that can be output with the error to help diagnose the fault?